src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -299,8 +305,12 @@ private void processData(InputData data) {
 				if (state != null) {
 					try {
 						State stateModel = gson.fromJson(state, State.class);
-						dataModel.latestState
-							.put(record.serialNumber, stateModel);
+						dataModel.latestStates
+							.computeIfAbsent(
+								record.serialNumber,
+								k -> new ArrayList<>()
+							)
+							.add(stateModel);
 						stateUpdates.add(record.serialNumber);
```

Comment:
Similar to how for wifiscans, we use `params.wifiScanBufferSize`, can we create and use something like `params.stateBufferSize` and make sure not to exceed the buffer size?

Reply:
I mean a new state buffer size is needed. We don't want to store all states and never delete them!